spec/controllers/comments_controller_spec.rb

```diff
@@ -0,0 +1,14 @@
+require 'spec_helper'
+
+describe CommentsController, type: :controller do
+  describe "POST #create" do
+    before do
+      @project = FactoryGirl.create(:project)
+      sign_in(@project.user)
+    end
+    it 'Response code is 200' do
+      xhr :post, :create, comment: { polycomment_type: "project", polycomment_id: 3, issue: false, project_name: @project.name, body: "heads are cool" }
```

Comment:
I don't know where this was broken, but looks like it's completely messed up. 

The whole point of `polycomment_type` and `polycomment_id` is that the former should refer to a class, and the latter should refer to the `id` of an object of that class. So in this case, `polycomment_id` should be set to a `project`'s id. Looks like we're using project_name here instead, to find the project - 

```
Project.find_by(name: params[:comment][:project_name])
```

We should also be checking `polycomment_id` to make sure that the object being commented on exists. This'll help to delete all dependent comments once their parent is deleted. 

Steps to do here - 

- Restore the use of `polycomment_id` and `polycomment_type`. 
- Remove `project_name`. 
- Create tests for an invalid `polycomment_id`, make sure a comment object is not saved to the database. 
- In this test, don't only check for a `200`, also make sure that the comment has been created in the database.

Extras: 
- Rename `polycomment` to `commentable` and use Rail's polymorphic associations. This way, we'll be able to access the object being commented on as `comment.commentable`. It'll automatically fetch a project or issue, whichever is appropriate.